python/ray/data/_internal/lazy_block_list.py

```diff
@@ -566,25 +613,58 @@ def _num_computed(self) -> int:
         return i
 
 
-def _execute_read_task(
+def _execute_read_task_nosplit(
     i: int,
     task: ReadTask,
     context: DatasetContext,
     stats_uuid: str,
     stats_actor: ray.actor.ActorHandle,
-) -> Tuple[MaybeBlockPartition, BlockPartitionMetadata]:
+) -> Tuple[Block, BlockMetadata]:
     DatasetContext._set_current(context)
     stats = BlockExecStats.builder()
 
     # Execute the task.
-    block = task()
+    block = next(task())
```

Comment:
Should it iterate over the whole `task()` instead of calling `next()` only once?

Reply:
I think we're guaranteed that the read task will only return a single block if dynamic block splitting is not turned on; however, even if that's the case, it might be a good idea to assert that the task return only contains a single block.